### ISSUE TITLE
Documentation lacks info about compiling / cleaning the db

### ISSUE THREAD

**Issue by mfornasa** [superscriptjs/superscript#151]
The documentation does not explain how to use `parse` and how to clean the db from existing gambits (see https://github.com/silentrob/superscript/issues/121 comments for example).

This make things _a lot harder_ for a beginner, as it lacks a suitable `HelloWord`. I spent an hour only to figure out how to run the bot and use the script samples in the wiki.

**Comment by mariusursache** [comment]
@mfornasa can you update the wiki with your findings, please?

For me, the easiest way to clean DBs are:
* Gambits/topics/users: just use a Mongo GUI such as Robomongo to remove stuff you want (or remove/drop manually).
* User facts in levelDB: delete the `/systemDB` folder.

**Comment by semateos** [comment]
I couldn't find this in the documentation either.  There is a script in bin called `cleanup.js` that refreshes the database.  With a global superscript install, that script ended up in my path.  So while editing scripts and testing, I can run `parse -f && cleanup` to purge the db and load the new content.  Then, if you want to get fancy you can watch for changes to the chat dir and run that.  And, since all of the chat dialog is in mongo, you can refresh the content without restarting the node bot server.

**Comment by mr-velamuri** [comment]
Seriously, It is very difficult to figureout many things in SuperScript. The documentation provided is at very high level. I am new to NodeJS and new to Superscript, with the available document I am able to implement a basic bot but to implement complex things it is very difficult to figure out how to implement them. 

It is mentioned in the documentation that we can save and get user variables using ^save and ^get methods. But it is no where mentioned that what I have to do if I have to get those values in a plugin.
say forexample, 
I have a gambit like this,

+ [My ticket type is ] *1
- ^save("ticketType", <cap>) Okay will do the need full 

then later in the conversations, I want to retrieve this ticket information in my plugin and want to process some logic based on its type

exports.myFunction = function myFunction(cb){
  if(^get("ticketType") == "category1"){
     //Do something
   }
}

If I do this way it is throwing error for ^ symbol. I tried multiple ways like this.get/this.user.get/this.user.db.get... nothing is working. 

Is there a way to achieve this or is it not possible. 

I request the authors to update the documentation with all necessary things..

**Comment by ZaDarkSide** [comment]
Actually it's documented here https://github.com/superscriptjs/superscript/wiki/Plugins-and-Functions

**Comment by mr-velamuri** [comment]
Yes it is documented. But it is at high level. If there any examples for each thing then it would be great.

It says 

" this.user

The user who sent the message as we know them. This will allow us to see the past messages in their history, how many conversation replies we have had, and when the conversation started."

But it is no where mentioned that how to retrieve those values from this object. is it this.user.get("ticketType")? or this.user.memory.db.get("ticketType") or this.user.memory.db.ticketType?

Not sure what notation to be used

**Comment by ZaDarkSide** [comment]
From the save/get plugin source:

```js
export function get(key, cb) {
	const memory = this.user.memory;
	const userId = this.user.id;

	debug('getVar', key, userId);

	memory.db.get({subject: key, predicate: userId}, (err, res) => {
		if (res && res.length !== 0) {
			cb(err, res[0].object);
		} else {
			cb(err, '');
		}
	});
}
```

**Comment by antwnic4** [comment]
Probably the easiest way to access variables, from the plugins functions, saved using `^save('test', test)` is: 
```
exports.testFunction= function testFunction(cb) {
  this.user.getVar('test', (err, res) => {
  console.log(res); 
  // => test
  cb(err, res);
  });
}
```

And saving vars using `setVar('test', test, cb);` respectively.

**Comment by mr-velamuri** [comment]
Thank so much both of you for timely help :) I appreciate it

**Comment by mr-velamuri** [comment]
I am back again :)

either the  this.user.memory.db.get or the getVar are assigning/sending this value to a callback function which is then by default can be used in gambit.

But my question here is how can I capture/return that value to make use in conditional statements?

something like this,

exports.myFunction = function myFunction(cb){
    
if(^get("ticketType") == "category1"){

           //Do something
      }
 }

This is my problem 

> if(^get("ticketType") == "category1")

**Comment by Lewwwk** [comment]
@mr-velamuri this is pretty off topic from the issue. If you are looking for general help, you might want to consider [joining the community](http://superscript-slackin.herokuapp.com). 

Within plugins you have access to both the user object and the message object so you should be able to perform whatever calculations you want.  You could also very simply integrate with any other code/db/collection you need to and even get a response using `this.bot.topicRedirect(user.id, messageTopic, gambit, callback)` from within a plugin if you wanted to get other script based responses from within a plugin.
